aiida/repository/backend/sandbox.py

```diff
@@ -16,11 +15,14 @@ class SandboxRepositoryBackend(AbstractRepositoryBackend):
     """Implementation of the ``AbstractRepositoryBackend`` using a sandbox folder on disk as the backend."""
 
     def __init__(self):
-        self._sandbox = None
+        from aiida.common.folders import SandboxFolder
+        self._sandbox: typing.Optional[SandboxFolder] = None
 
     def __str__(self) -> str:
         """Return the string representation of this repository."""
-        return f'SandboxRepository: {self._sandbox.abspath}'
+        if self.is_initialised:
+            return f'SandboxRepository: {self._sandbox.abspath if self._sandbox else "null"}'
```

Comment:
Minor nitpick alert: the conditional in this line is no longer necessary since it is couched in `if self.is_initialised`.

Reply:
It's still required for mypy though

Reply:
ugh... when tools that are supposed to help get in the way. Fine to keep as is then